Lunch/ViewController.swift

```diff
@@ -58,12 +92,148 @@ class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSour
     }
 
     func textFieldDidEndEditing(textField: UITextField) {
-        items.append(self.lunchTextField.text)
-        var indexSet = NSIndexSet(index: 0)
-        lunchTableView.beginUpdates()
-        self.lunchTableView.reloadSections(indexSet, withRowAnimation: UITableViewRowAnimation.Fade)
-        lunchTableView.endUpdates()
+        let place = self.lunchTextField.text
+        let vote = 0
+        eateries.append(place)
+        votes.append(vote)
+        let testObject = PFObject(className: classNameKey)
+        testObject[placeColumnKey] = place
+        testObject[voteColumnKey] = vote
+        testObject.saveInBackgroundWithBlock {
+            (success: Bool, error: NSError?) -> Void in
+            if success {
+                println("Place \(place) has been saved with \(vote) votes")
+            }
+        }
+        updateTableView()
         self.lunchTextField.text = emptyString
+        let time = 60.0
+        let timer = NSTimer.scheduledTimerWithTimeInterval(time, target: self, selector: "startVoting", userInfo: nil,repeats: false)
+    }
+
+    func startVoting() {
+        if !votePushSent {
+            votePushSent = true
+            let push = PFInstallation.query()
+            push?.whereKey("deviceType", equalTo: "ios")
+            PFPush.sendPushMessageToQueryInBackground(push
+                , withMessage: "Start voting! \nYou have 30 minutes until voting ends")
+            let time = 30.0 * 60.0
+            let timer = NSTimer.scheduledTimerWithTimeInterval(time, target: self, selector: "stopVoting", userInfo: nil, repeats: false)
```

Comment:
Yeah you can't do this client-side, the app probably won't even be running half an hour later